openeo/rest/connection.py

```diff
@@ -41,20 +41,18 @@ def url_join(root_url: str, path: str):
     return urljoin(root_url.rstrip('/') + '/', path.lstrip('/'))
 
 
-class OpenEoApiError(OpenEoClientException):
-    """
-    Error returned by OpenEO API according to https://open-eo.github.io/openeo-api/errors/
-    """
-
-    def __init__(self, http_status_code: int = None,
-                 code: str = 'unknown', message: str = 'unknown error', id: str = None, url: str = None):
-        self.http_status_code = http_status_code
-        self.code = code
-        self.message = message
-        self.id = id
-        self.url = url
-        super().__init__("[{s}] {c}: {m}".format(s=self.http_status_code, c=self.code, m=self.message))
-
+def paginate(con, url: str, params: dict = {}, callback: Callable = lambda resp, page: resp):
```

Comment:
do you have experience with unit tests and pytest @m-mohr ? I think this one deserves a bit of coverage.
If not, I can have a look after merging

Reply:
Unit Tests yes, pytest no.

Reply:
we use a http mock library (requests_mock) for fake url serving, so we don't need a real "online" endpoint

Reply:
Good.
To get things in so that others can test things for the SRR, I'd rather postpone to later. If you don't have the time, I can give it a try, maybe we can just open an issue so that we don't forget.

Reply:
https://github.com/Open-EO/openeo-python-client/issues/186